### ISSUE TITLE
Fullscreen mode does not work on Safari Mobile on iOS

### ISSUE THREAD

**Issue by jhemmje** [maplibre/maplibre-gl-js#3306]
**maplibre-gl-js version**: 3.5.2

**browser**: Safari Mobile on iOS. Tested on 1) ❌ iPhone Simulator (*iOS versions 17.0, 16.4 & 15.8*), 2) ✅ iPad Simulator (*iOS version 17.0*) (*Not tested on a real iPhone, I do not have one*)

### Steps to Trigger Behavior
 1. Navigate to https://maplibre.org/maplibre-gl-js-docs/example/fullscreen/ using an iPhone.
 2. Click on the `Enter fullscreen` icon.
 3. On iPhones the icon toggles, but the fullscreen mode is not entered. On iPads the fullscreen mode works as expected.

### Link to Demonstration
https://maplibre.org/maplibre-gl-js-docs/example/fullscreen/

![Fullscreen-mode](https://github.com/maplibre/maplibre-gl-js/assets/113904/a91f0e4d-f468-4490-9648-1920532d5ee3)

### Expected Behavior
Fullscreen mode works on iPhone devices.

### Actual Behavior
Fullscreen mode is not entered on iPhone devices.

### Further notes
* The problem is not reproducible using Safari's Enter Responsive Design Mode as it does not simulate a device.
* Safari Developer mode has a way to enable the JavaScript Console on a Simulator or real device. But no error or warning is shown.
* https://github.com/maplibre/maplibre-gl-js/issues/1919 is related to this issue.

**Comment by HarelM** [comment]
I would try and see if there's a problem on a real device, if not, then it's a simulator bug, which is not so interesting from my point of view...

**Comment by jhemmje** [comment]
I tested the [fullscreen example page](https://maplibre.org/maplibre-gl-js/docs/examples/fullscreen/) using [TestingBot](https://testingbot.com/mobile/realdevicetesting) on a real iPhone and it also the same problem. So it is not a simulator bug.

However I have continued testing with the iPhone Simulator and found out the following:
* A pseudo-fullscreen mode is used on iPhones, see https://github.com/maplibre/maplibre-gl-js/pull/1678.
* In this mode the map is enlarged using `height: 100%` and `width: 100%` via the CSS class `maplibregl-pseudo-fullscreen`. The navigation bar of Safari is still visible, hence the term pseudo-fullscreen.
* This does not work on the example page, obviously because the map is embedded inside an iFrame. Within the iFrame, the map is already at full size, so nothing happens when you click on the fullscreen button.
* If I copy and paste the example code to a file and open it on the iPhone, the fullscreen mode works as expected.
* The good news is, that this is not a bug inside MapLibre GL JS. However it's confusing that the example page is not working on iPhones because of the iFrame. I think that a note on the example page would be helpful so that others don't stumble across the same problem. Or maybe there is a better way for embedding the example maps in the documentation.
